aspects/request_timeout_handler_test.go

```diff
@@ -0,0 +1,57 @@
+package aspects
+
+import (
+	"github.com/julienschmidt/httprouter"
+	"github.com/prebid/prebid-server/config"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const reqTimeInQueueHeaderName = "X-Ngx-Request-Time"
+const reqTimeoutHeaderName = "X-Ngx-Request-Timeout"
+
+func TestQueuedRequestTimeoutWithTimeout(t *testing.T) {
+
+	rw := ExecuteAspectRequest(t, "6")
+
+	assert.Equal(t, http.StatusRequestTimeout, rw.Code, "Http response code is incorrect, should be 408")
+
+}
+
+func TestQueuedRequestTimeoutNoTimeout(t *testing.T) {
+
+	rw := ExecuteAspectRequest(t, "0.9")
+
+	assert.Equal(t, http.StatusOK, rw.Code, "Http response code is incorrect, should be 200")
+
+}
+
+func MockEndpoint() httprouter.Handle {
+	return httprouter.Handle(MockHandler)
+}
+
+func MockHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {}
+
+func ExecuteAspectRequest(t *testing.T, timeInQueue string) *httptest.ResponseRecorder {
+	rw := httptest.NewRecorder()
+	req, err := http.NewRequest("POST", "/test", nil)
+	if err != nil {
+		assert.Fail(t, "Unable create mock http request")
+	}
+	req.Header.Set(reqTimeInQueueHeaderName, timeInQueue)
```

Comment:
Also add test cases where the headers are not given or are not valid numbers to make sure we don't fail bad if we get unexpected values here.

Reply:
Yes, added